File: acmp.ru/[0001] - [0100]/[0026] Две окружности.py
from math import sqrt

x1, y1, r1 = [int(i) for i in input().split()]
x2, y2, r2 = [int(i) for i in input().split()]

k = sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)

print('YES' if k <= r1 + r2 and k >= abs(r1 - r2) else 'NO')

File: acmp.ru/[0001] - [0100]/[0026] Две окружности.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int x1, y1, r1, x2, y2, r2;
    float k;

    cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;

    k = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));

    cout << (k <= r1 + r2 && k >= abs(r1 - r2) ? "YES" : "NO") << endl;

    return 0;
}
